app/controllers/mixins/custom_button_dialog_form_mixin.rb

```diff
@@ -0,0 +1,32 @@
+module Mixins
+  module CustomButtonDialogFormMixin
+    def set_custom_button_dialog_presenter
+      presenter ||= ExplorerPresenter.new(
+        :active_tree => x_active_tree,
+      )
+      presenter.show(:default_left_cell).hide(:custom_left_cell)
+      presenter.update(:main_div, r[:partial => "shared/dialogs/dialog_provision"])
+      if @record.try(:dialog_fields)
+        @record.dialog_fields.each do |field|
```

Comment:
This is strange. You look over ALL fields. And for every field of certain types you do
```
   presenter[:build_calendar] = { :date_from => field.show_past_dates ? nil : Time.zone.now }
```
So if thre are multiple fields you are setting that multiple times.
And if one field has  `field.show_past_dates` `true` and another `false` than whatever comes last wins

That is really strange.

Is that intended? That does it do?

Reply:
... this is also present in `ServiceController`, `CatalogController` and `VmCommon` - all in `#replace_right_cell` -
 should probably fix this for all of them.

Reply:
`miqBuildCalendar` [supports](https://github.com/ManageIQ/manageiq-ui-classic/blob/d0f08dae929eeec93467e3cf2422dbe6b786f119/app/assets/javascripts/miq_application.js#L860) `data_date_start` directly on the element.

So you should be able to drop this bit of code in favour of some addition to the view..

Reply:
*views...

```
app/views/miq_ae_customization/_dialog_sample.html.haml
35:                      - when "DialogFieldDateControl", "DialogFieldDateTimeControl"

shared/dialogs/dialog_field_date_and_date_time_control
1:- if field.type == "DialogFieldDateControl"
7:- if field.type == "DialogFieldDateTimeControl"
```